packages/compass-aggregations/src/components/pipeline-explain/explain-results.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { css, spacing, Card } from '@mongodb-js/compass-components';
+import { Document } from '@mongodb-js/compass-crud';
+import HadronDocument from 'hadron-document';
+
+import type { ExplainData } from '../../modules/explain';
+import { ExplainQueryPerformance } from './explain-query-performance';
+
+type ExplainResultsProps = {
+  plan: ExplainData['plan'];
+  stats?: ExplainData['stats'];
+};
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[4],
+});
+
+const cardStyles = css({
+  height: '350px',
+  overflowY: 'scroll',
+});
+
+export const ExplainResults: React.FunctionComponent<ExplainResultsProps> = ({
+  plan,
+  stats,
+}) => {
+  const doc = new HadronDocument(plan);
+  return (
+    <div className={containerStyles} data-testid="pipeline-explain-results">
+      {stats && (
+        <ExplainQueryPerformance
+          nReturned={stats.nReturned}
+          executionTimeMillis={stats.executionTimeMillis}
+          usedIndexes={stats.usedIndexes}
+        />
+      )}
+      <Card className={cardStyles} data-testid="pipeline-explain-results-json">
+        <Document
+          doc={doc}
```

Comment:
As we recently discovered re-creating `HadronDocument` in render [is really not great](https://github.com/mongodb-js/compass/pull/3073) so instead of passing a document instance created on every render, you can pass the object that the document is based on instead

```suggestion
          doc={plan}
```

The memoization of the `HadronDocument` instance will be handled by the `Document` component itself